eva/executor/create_udf_executor.py

```diff
@@ -60,9 +63,19 @@ def exec(self):
             )
             logger.error(err_msg)
             raise RuntimeError(err_msg)
-        catalog_manager.create_udf(
+
+        # create the actual udf
+        udf_metadata = catalog_manager.create_udf(
             self.node.name, impl_path, self.node.udf_type, io_list
         )
+
+        # Profile the UDF
+        profiler = Profiler(impl_path, self.node.name)
+        metrics = profiler.run()
+
+        # Insert the profiled UDF to catalog
+        catalog_manager.create_udf_profile(udf_metadata.id, metrics)
```

Comment:
Profiler is blocking. TODO: Making it non-blocking

This is non-trivial: If non-blocking, potential resources are competed causing incorrect measurement of time.

Having said that, during actual execution of query as well, resources could be competed. Adaptive optimizer is the solution for this. 

NOTE: We are not intending to make it non-blocking in the scope of the project.